lib/activity.rb

```diff
@@ -0,0 +1,34 @@
+class Activity
+
+  attr_reader :name,
+              :participants,
+              :total_cost,
+              :owed
+
+  def initialize(name)
+    @name = name
+    @participants = Hash.new(0)
+  end
+
+  def add_participant(name, paid)
+    @participants[name] += paid
+  end
+
+  def total_cost
+    @participants.map do |person,paid|
```

Comment:
Good job iterating over this hash!  You could also have done something like this `@participants.values.sum`